packages/flutter_test/test/widget_tester_test.dart

```diff
@@ -751,6 +751,23 @@ void main() {
       expect(desktop.values.union(mobile.values), equals(all.values));
     });
   });
+
+  group('no timers pending after widget tree disposal', () {
```

Comment:
I think you probably want to check that the assert exception just includes the assert message.

Reply:
What is the proper way of doing this? I checked by creating a timer and not cancelling it. I see that the code is no longer printed below, but I don't know how to include a unit test for this without causing the `Some tests failed` message due to the actual assert error message needing to be printed.
```
Pending timers:
Timer (duration: 0:00:01.000000, periodic: false), created:
#0      new FakeTimer._ (package:fake_async/fake_async.dart:283:41)
#1      FakeAsync._createTimer (package:fake_async/fake_async.dart:247:27)
#2      FakeAsync.run.<anonymous closure> (package:fake_async/fake_async.dart:180:19)
#5      main.<anonymous closure>.<anonymous closure> (file:///Users/christiangonzalez/Desktop/flutter/packages/flutter_test/test/widget_tester_test.dart:764:27)
#6      testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart:144:29)
<asynchronous suspension>
#7      testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart)
#8      TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:703:19)
<asynchronous suspension>
#11     TestWidgetsFlutterBinding._runTest (package:flutter_test/src/binding.dart:683:14)
#12     AutomatedTestWidgetsFlutterBinding.runTest.<anonymous closure> (package:flutter_test/src/binding.dart:1083:24)
#13     FakeAsync.run.<anonymous closure>.<anonymous closure> (package:fake_async/fake_async.dart:177:54)
#18     withClock (package:clock/src/default.dart:46:10)
#19     FakeAsync.run.<anonymous closure> (package:fake_async/fake_async.dart:177:22)
#24     FakeAsync.run (package:fake_async/fake_async.dart:177:7)
#25     AutomatedTestWidgetsFlutterBinding.runTest (package:flutter_test/src/binding.dart:1080:15)
#26     testWidgets.<anonymous closure> (package:flutter_test/src/widget_tester.dart:137:24)
#27     Declarer.test.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart:172:27)
<asynchronous suspension>
#28     Declarer.test.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart)
#29     Invoker.waitForOutstandingCallbacks.<anonymous closure> (package:test_api/src/backend/invoker.dart:246:15)
#34     Invoker.waitForOutstandingCallbacks (package:test_api/src/backend/invoker.dart:243:5)
#35     Declarer.test.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart:170:33)
#40     Declarer.test.<anonymous closure> (package:test_api/src/backend/declarer.dart:169:13)
#41     Invoker._onRun.<anonymous closure>.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/invoker.dart:400:30)
(elided 36 frames from class _RawReceivePortImpl, class _Timer, dart:async, dart:async-patch, and package:stack_trace)

══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following assertion was thrown running a test:
A Timer is still pending even after the widget tree was disposed.
'package:flutter_test/src/binding.dart':
Failed assertion: line 1134 pos 12: '!timersPending'

When the exception was thrown, this was the stack:
#2      AutomatedTestWidgetsFlutterBinding._verifyInvariants (package:flutter_test/src/binding.dart:1134:12)
#3      TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:716:7)
<asynchronous suspension>
#6      TestWidgetsFlutterBinding._runTest (package:flutter_test/src/binding.dart:683:14)
#7      AutomatedTestWidgetsFlutterBinding.runTest.<anonymous closure> (package:flutter_test/src/binding.dart:1083:24)
#8      FakeAsync.run.<anonymous closure>.<anonymous closure> (package:fake_async/fake_async.dart:177:54)
#13     withClock (package:clock/src/default.dart:46:10)
#14     FakeAsync.run.<anonymous closure> (package:fake_async/fake_async.dart:177:22)
#19     FakeAsync.run (package:fake_async/fake_async.dart:177:7)
#20     AutomatedTestWidgetsFlutterBinding.runTest (package:flutter_test/src/binding.dart:1080:15)
#21     testWidgets.<anonymous closure> (package:flutter_test/src/widget_tester.dart:137:24)
#22     Declarer.test.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart:172:27)
<asynchronous suspension>
#23     Declarer.test.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart)
#24     Invoker.waitForOutstandingCallbacks.<anonymous closure> (package:test_api/src/backend/invoker.dart:246:15)
#29     Invoker.waitForOutstandingCallbacks (package:test_api/src/backend/invoker.dart:243:5)
#30     Declarer.test.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart:170:33)
#35     Declarer.test.<anonymous closure> (package:test_api/src/backend/declarer.dart:169:13)
#36     Invoker._onRun.<anonymous closure>.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/invoker.dart:400:30)
(elided 36 frames from class _AssertionError, class _RawReceivePortImpl, class _Timer, dart:async, dart:async-patch, and package:stack_trace)

The test description was:
  timers pending
```

Reply:
you could have a something like:
```dart
expect(
  () { 
    // Put code that throws here.
  },
  throwsA(predicate<AssertionError>((AssertionError e) => e.message == '<the-message>')),
);
```